packages/engine/paima-runtime/src/cde-config/utils.ts

```diff
@@ -1,15 +1,31 @@
 import * as fs from 'fs/promises';
 
-import { doLog } from '@paima/utils';
+import { ChainDataExtensionType, doLog } from '@paima/utils';
 
 import type { ChainDataExtension } from '@paima/sm';
 
 export function getEarliestStartBlockheight(config: ChainDataExtension[]): number {
-  const startBlockheights = config.map(cde => cde.startBlockHeight).filter(sbh => !!sbh);
+  const startBlockheights = config.reduce((arr, cde) => {
+    if (cde.cdeType !== ChainDataExtensionType.CardanoPool) {
+      arr.push(cde.startBlockHeight);
+    }
+    return arr;
+  }, [] as number[]);
   const minStartBlockheight = Math.min(...startBlockheights);
   return isFinite(minStartBlockheight) ? minStartBlockheight : -1;
```

Comment:
we can put more in the reduce. There isn't a point in building an array just to call `Math.min` on it afterwards (instead of just having the reduce find the minimum)